machine_learning_hep/analysis/analyzerdhadrons_mult.py

```diff
@@ -647,28 +648,31 @@ def do_plot(histo):
             def plot_tpc_tof_me(tag):
                 # Compute TPC-TOF matching efficiency
                 if tpc_tof_me:
-                    for i in ["Pi", "K", "Pr"]:
-                        for j in ["0", "1"]:
-                            for k in ["p", "pt"]:
-                                hname = [f"{k}_prong{j}",
-                                         f"nsigTOF_{i}_{j}", tag]
-                                hnum = get_histo(*hname,
-                                                 strictly_require=False)
-                                if hnum is None:
-                                    continue
-                                hnum = hnum.ProjectionX(
-                                    hnum.GetName() + "_num", 2, -1)
-                                hden = get_histo(*hname)
-                                hden = hden.ProjectionX(
-                                    hden.GetName() + "_den")
-                                hnum.Divide(hnum, hden, 1, 1, "B")
-                                hnum.SetName(
-                                    hnum.GetName().replace(
-                                        "_num", "_TPC-TOF_MatchingEfficiency"
-                                    )
-                                )
-                                hnum.GetYaxis().SetTitle("TPC-TOF_MatchingEfficiency")
-                                do_plot(hnum)
+                    to_plot = [["Pi", "K", "Pr"],
+                               ["0", "1"],
+                               ["p_prong0", "pt_prong0", "pt_cand"]
+                               ]
+                    to_plot = list(itertools.product(*to_plot))
+                    for spec, prong, observable in to_plot:
```

Comment:
Hej @njacazio 
just one thing: You should be able to directly do
```python
for spec, prong, observable in itertools.product(*to_plot):
  # ...
```

Reply:
Agreed, done!